Kernel_Documentation/Evaluating_and_improving_the_Linux_Kernel_documentation.md

```diff
@@ -0,0 +1,107 @@
+
+# Evaluating and improving the Linux Kernel documentation
+
+
+# Goal and Rationale
+
+The goal of this document is to analyze the current templates and guidelines that are available in the [Linux Kernel documentation](https://docs.kernel.org/), evaluate if and how they fulfill architecture and design aspects required by functional safety, define improvements also in consideration of maintenance challenges deriving from a continuously evolving code baseline.
+
+The main reason for such an analysis is to improve the baseline of available documentation evidence for SW Architectures and Design aspects, starting from the process of documenting these, in a way that can simplify the adoption of Linux in a functional safety system.
```

Comment:
They are intentionally not in the scope of this document. When it comes to pre-existing code it is way less ambiguous to capture the expected behaviour as "SW Architecture and Design doc" rather than requirements.
The main issue I see is that, whether a requirement is correct or not can be highly arguable depending on the context of integration, whereas an existing code architecture and design must be accepted for what it is. The in turn it is the integrator that must assess the adequacy of the architecture and design against the requirements allocated by him to Linux

Reply:
Ok. Makes sense. Thanks for the explanation.

Reply:
I understand your point and make sense to avoid a sw requirements reverse engineering for complex pre existing software. But it is also true that software is in continuous evolution and we cannot exclude that in a few years it will be completely re-designed. In that regard having a template able to support software requirements can help developers to start any new development from the requirements design.
Moreover that seems to me a great and unique opportunity to make some change and drive the project to a new development paradigma, requirements based.

Reply:
@pellecchialuigi  Linux development is code driven and not requirement driven. So I do not think that we can enforce a requirements driven development, plus being Linux a multipurpose OS the requirements allocated to it can hugely vary depending on the HW platform and on the specific use cases.
Hence I still think that it is better to document the SW Architecture and Design of the code and the same can stay up to date as the code evolves

Reply:
Also I think that it is better to start documenting aspects that are closer to the code (such as design and architecture) and later expand, also with requirements, if the Linux community realises that we need these. I.e. I do not want to kill a fire by putting too much wood on it.

Reply:
@pellecchialuigi  are you ok with this resolution ^^^ ?

Reply:
Yes, make sense. The community will chose the way.
Let's start the way that is closest to the current situation. Thanks for your clarification and feedback.